lib/AST/GenericSignature.cpp

```diff
@@ -1344,6 +1344,12 @@ void swift::simple_display(raw_ostream &out, GenericSignature sig) {
     out << "NULL";
 }
 
+Requirement::Requirement(RequirementKind kind, Type first, Type second)
+    : RequirementBase(kind, first, second) {
+  assert(!(kind == RequirementKind::Conformance &&
```

Comment:
IIRC we create conformance requirements with arbitrary types on the right hand side for invalid code like `func foo<T : Int>(_: T)`. I wonder what will happen if you write `func foo<T : any P>(_: T)`, will it hit this assertion? A better place to check generic signature invariants might be in GenericSignature::verify(), which checks minimal/canonical requirements that are output by the GSB.

Reply:
Oh of course, you're totally right. I removed the assertion and instead fixed this in `GenericSignatureBuilder::addTypeRequirement` which is where the current check for non-protocol, non-class constraints is. I also added a test for `<T: any P>`. Thanks!